### PR TITLE
Update and tidy `go.mod`

### PR DESCRIPTION
**Pull request by adlerjohn**
- Tidy `go.mod`
- ~~Remove unused dependencies~~
- Update dependencies
- Update Go version to 1.15

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,17 +1,21 @@
 module github.com/lazyledger/rsmt2d
 
-go 1.14
+go 1.15
 
 require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/kr/text v0.2.0 // indirect
```
Comment (liamsi) [inline go.mod:7]: Can we at least understand what caused this to be imported? If it is gonum, we will remove gonum with #22.

File: go.mod [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
 	// only needed if built with go build -tags leopard
-	github.com/lazyledger/go-leopard v0.0.0-20200604113236-298f93361181
+	github.com/lazyledger/go-leopard v0.0.0-20200724211609-50ec4b3fab41
```
Comment (liamsi) [inline go.mod:9]: We should also tag go-leopard. I'll revisit that repository soon.

File: go.mod [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
-	github.com/spacemonkeygo/errors v0.0.0-20171212215202-9064522e9fd1 // indirect
-	github.com/stretchr/testify v1.5.1
-	github.com/vivint/infectious v0.0.0-20190108171102-2455b059135b
-	golang.org/x/net v0.0.0-20200520182314-0ba52f642ac2 // indirect
-	golang.org/x/sys v0.0.0-20200519105757-fe76b779f299 // indirect
-	gonum.org/v1/gonum v0.7.0
-	gopkg.in/yaml.v2 v2.3.0 // indirect
+	github.com/stretchr/testify v1.7.0
+	github.com/vivint/infectious v0.0.0-20200605153912-25a574ae18a3
+	gitlab.com/NebulousLabs/errors v0.0.0-20200929122200-06c536cf6975 // indirect
```
Comment (liamsi) [inline go.mod:13]: It's really surprising to see these updated while the merkle tree stayed the same 🤔

**Review by liamsi — APPROVED** [review]
👍🏼

File: go.mod (hunk 1 above)
Comment (adlerjohn) [reply go.mod:7]: For posterity: this is dependency of `yaml` v3, which is a dependency of `testify`

**Pull request merged**